src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -12,29 +12,35 @@ namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public class WebJobsScriptHostService : IHostedService, IDisposable
     {
+        private readonly ILinuxContainerInitializationService _linuxContainerInitializationService;
         private readonly WebScriptHostManager _scriptHostManager;
         private readonly CancellationTokenSource _cancellationTokenSource;
         private readonly ILogger _logger;
         private bool _disposed = false;
         private Task _hostTask;
 
-        public WebJobsScriptHostService(WebScriptHostManager scriptHostManager, ILoggerFactory loggerFactory)
+        public WebJobsScriptHostService(WebScriptHostManager scriptHostManager, ILinuxContainerInitializationService linuxContainerInitializationService, ILoggerFactory loggerFactory)
         {
             _scriptHostManager = scriptHostManager ?? throw new ArgumentException($@"Unable to locate the {nameof(WebScriptHostManager)} service. " +
                     $"Please add all the required services by calling '{nameof(IServiceCollection)}.{nameof(WebJobsServiceCollectionExtensions.AddWebJobsScriptHost)}' " +
                     $"inside the call to 'ConfigureServices' in the application startup code");
 
             _cancellationTokenSource = new CancellationTokenSource();
             _hostTask = Task.CompletedTask;
+            _linuxContainerInitializationService = linuxContainerInitializationService;
             _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
         }
 
-        public Task StartAsync(CancellationToken cancellationToken)
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
+            _logger.LogInformation("Initializing Container.");
+
+            // Do additional initialization if running inside linux container.
+            await _linuxContainerInitializationService.Run(cancellationToken);
```

Comment:
Can we move this initialization down to the host initialization? Or, if needed, to its own hosted service? I would like to keep this service simple and avoid expanding the scope overtime.

Reply:
Moved it to its own hosted service.